PartsApp/ContragentOperationsInfoForm.cs

```diff
@@ -120,24 +145,81 @@ private void ContragentsListBox_MouseDown(object sender, MouseEventArgs e)
         }
 
         private void EditContragentToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            var contragent = GetSelectedContragentFromDB();
+
+            //Передаём в форму 'свежую'инф-цию из базы, на случай если она обновилась.
+            new AddContragentForm(contragent).Show();            
+        }
+
+        private void OnDisableOrEnableContragentToolStripMenuItemClick(object sender, EventArgs e)
+        {
+            var contragent = GetSelectedContragentFromCache();
+            var input = contragent.IsActive ? ShowDisableContragentMessageBox(contragent) : ShowEnableContragentMessageBox(contragent);
+
+            if (input == DialogResult.Yes)
+            {               
+                try
+                {
+                    contragent.IsActive = !contragent.IsActive;
+                    PartsDAL.UpdateContragent(contragent);
```

Comment:
No. That's not how we do it.
Enabling/disabling something is big enough functionality to be placed into its own method.
Typically we have a business layer method 'Disable' and 'Enable' separately because many additional actions may need to be done, like sending an email or removing from other, 3rd party systems.
In our case there is no need for a complex flow. So we can just have two separate methods in the repo, like:  void Disable(int id), and the same for Enable

Reply:
added Disable / Enable methods to Customer and Supplier repos